### ISSUE TITLE
(libretro) Problems with Disk Control

### ISSUE THREAD

**Issue by orbea** [AZO234/NP2kai#69]
OS: `Slackware64-currnet`
NP2kai: https://github.com/AZO234/NP2kai/commit/e0b9fe55279e5d52df71c070a347c30ab3bf5c8b
RetroArch: `1.7.7`

I am trying the English translation of the game `Farland Story` as can be seen in the following link.

https://www.romhacking.net/translations/646/

This contains four `fdi` files.
```
Farland Story 1 (English)(TGL)(Disk A).fdi
Farland Story 1 (English)(TGL)(Disk B).fdi
Farland Story 1 (English)(TGL)(Disk C).fdi
Farland Story 1 (English)(TGL)(Disk D).fdi
```
If I load `Disk A` with the libretro core it loads correctly, but I get stuck when it asks me to change disks after starting a game, the `Disk Control` option in the RetroArch's quick menu is not available and I can not manually append disks. 

The `Disk Control` option is available if I open a single file with other libretro cores like `beetle-psx`.

![1](https://i.imgur.com/zEtW3Vh.png)

So I next added the four `fdi` files to a plain text `m3u` file.
```
$ cat Farland\ Story\ 1.m3u 
Farland Story 1 (English)(TGL)(Disk A).fdi
Farland Story 1 (English)(TGL)(Disk B).fdi
Farland Story 1 (English)(TGL)(Disk C).fdi
Farland Story 1 (English)(TGL)(Disk D).fdi
```
And if I launch this `m3u` file I notice a few issues.

1. I get stuck on this screen directly after the memory loading screen and the game never starts.
![2](https://i.imgur.com/E4DgHCm.png)

2. From stdout I notice its trying to load `Disk B` instead of `Disk A`, but even if I edit the `m3u` file to make `Disk A` the second item in the list it does not help.
```
$ retroarch -L np2kai_libretro.so Farland\ Story\ 1.m3u 
arg_0:np2kai
arg_1: "/Farland Story 1 (English)(TGL)(Disk B).fdi"
Samples:782
INIT done
```
Edit: The `disk control` option in RetroArch's quick menu is available if I load a `m3u` file, but it does not seem to help here.

Is there anything that can be done to make this game playable with the libretro `NP2kai` core?

**Comment by AZO234** [comment]
I have not investigated it yet, but I have some advice.

NP2kai has two FDDs and is not made to have only one FDD.
So, the first line of m3u mounts to FDD1, the second line mounts to FDD2, respectively.

"Option"-> 'Swap Disk on Drive' = FDD1
Have you tried the above settings?

About pic of after the memory loading screen.
It is displayed when the boot of FDD fails if the BIOS of PC-98 is not correctly installed in RetroArch's system directory.

Thanks to your report.

**Comment by orbea** [comment]
> "Option"-> 'Swap Disk on Drive' = FDD1
Have you tried the above settings?

I tried this setting now and it did not make a difference, the stdout when loading a `m3u` file still shows `Disk B`.

> About pic of after the memory loading screen.
It is displayed when the boot of FDD fails if the BIOS of PC-98 is not correctly installed in RetroArch's system directory.

Yes, you are right. I did not have the bios files. I now added them and it now brings me to this screen when loading and `m3u` file.

![3](https://i.imgur.com/bxM5wFH.png)

I do not understand why it proceeds to load the game without bios files if I load the `fdi` file directly, but fails if load a `m3u` file?
